perceiver/data/audio/symbolic.py

```diff
@@ -0,0 +1,216 @@
+import functools
+import multiprocessing
+import os
+import random
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+from perceiver.data.audio.midi_processor import encode_midi_files
+
+
+class SymbolicAudioDataModule(pl.LightningDataModule):
+    _EXAMPLE_SEPARATOR_INPUT_ID = -1
+    _PAD_INPUT_ID = 388
+    _VOCAB_SIZE = 389
+
+    def __init__(
+        self,
+        dataset_dir: str,
+        max_seq_len: int,
+        min_seq_len: Optional[int] = None,
+        padding_side: Optional[str] = "left",
```

Comment:
Use `None` as default for an optional parameter.

Reply:
I think in this case its preferable to drop the `Optional` type

Reply:
All other modules have `Optional` with default `None`. Shouldn't we prefer consistency now and do a library-wide change later if needed?

Reply:
In other modules (e.g. `TextDataModule`) it is acceptable to set the `padding_side` to `None`, as the default of the tokenizer is applied in this case. In this data module, we don't use a tokenizer with a set default, so we have to define the default value. We can set the default value manually in the constructor or in the `SymbolicAudioCollator` but I think setting the default here and consequently preserving the value in the `hparams` would be preferrable.

Reply:
:+1: Good point, I agree